doc/nrf/libraries/dfu/dfu_extra.rst

```diff
@@ -0,0 +1,107 @@
+.. _lib_dfu_extra:
+
+DFU extra image extensions
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The DFU extra image module provides a flexible mechanism for including extra firmware images in the DFU packages.
+This allows applications to extend the built-in DFU functionality with additional extra firmware images beyond ones supported natively in the nRF Connect SDK, for example firmware for external device.
+
+This CMake extension supports the following DFU delivery mechanisms:
+
+* **Multi-image binary** (``dfu_multi_image.bin``)
+
+  Multi-image binary is used by applications that require downloading the DFU package containing multiple images as a single blob. The application is then responsible for handling the images. Refer to :ref:`lib_dfu_multi_image` to see how to provide firmware implementation.
+
+* **ZIP packages** (``dfu_application.zip``)
+
+  ZIP packages are used for updates via SMP protocol. The firmware responsible for handling the images uses SMP image management which automatically handles DFU of up to 3 images (MCUboot slots). In case addition of yet another image is needed, user needs to adjust the ``zephyr_img_mgmt.c`` file. For more information about SMP, see :ref:`Device Management <zephyr:device_mgmt>` documentation.
```

Comment:
```suggestion
* ZIP packages (:file:`dfu_application.zip`)

  ZIP packages are used for updates through Simple Management Protocol (SMP). 
  The firmware responsible for handling the images uses SMP image management, which automatically handles DFU for up to three images (MCUboot slots). 
  In case you need to add another image, you must adjust the :fle:`zephyr_img_mgmt.c` file. 
  For more information about SMP, see the :ref:`Device Management <zephyr:device_mgmt>` documentation.
```